qiskit/utils/mitigation/__init__.py

```diff
@@ -0,0 +1,54 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# This code was originally copied from the qiskit-ignis repsoitory see:
+# https://github.com/Qiskit/qiskit-ignis/blob/b91066c72171bcd55a70e6e8993b813ec763cf41/qiskit/ignis/mitigation/measurement/__init__.py
+# it was migrated as qiskit-ignis is being deprecated
+
+"""
+=============================================================
+Measurement Mitigation Utils (:mod:`qiskit.utils.mitigation`)
+=============================================================
+
+.. currentmodule:: qiskit.utils.mitigation
+
+.. warning::
+
+    The user facing API stability of this module is not guaranteed except for
+    their use with the :class:`~qiskit.utils.QuantumInstance` (ie using the
+    :class:`~qiskit.utils.mitigation.CompleteMeasFitter` or
+    :class:`~qiskit.utils.mitigation.TensoredMeasFitter` class as values for the
+    ``meas_error_mitigation_cls``). The rest of this module should be treated as
+    an internal private API that can not be relied upon.
+
+Measurement correction
+======================
+
+The measurement calibration is used to mitigate measurement errors.
+The main idea is to prepare all :math:`2^n` basis input states and compute
+the probability of measuring counts in the other basis states.
+From these calibrations, it is possible to correct the average results
+of another experiment of interest. These tools are intended for use solely
+with the :class:`~qiskit.utils.QuantumInstance` class as part of
+:mod:`qiskit.algorithms` and :mod:`qiskit.opflow`.
+
+.. autosummary::
+   :toctree: ../stubs/
+
+   CompleteMeasFitter
+   TensoredMeasFitter
+"""
+
+# Measurement correction functions
+from .circuits import complete_meas_cal, tensored_meas_cal
+from .filters import MeasurementFilter, TensoredFilter
+from .fitters import CompleteMeasFitter, TensoredMeasFitter
```

Comment:
Yeah, that's not a bad idea. I was trying to avoid the `_module.py` pattern mostly because in the dark before times (i.e. before I first started working on Qiskit) pretty much every module was `_foo.py` and the user api was defined as the re-exported classes from the `__init__.py` (without `__all__`). See https://github.com/Qiskit/qiskit-terra/tree/0.5.0/qiskit if you'd like to get a more concrete feel for what I'm describing. It's left a bad taste ever since so I just try to avoid it so people aren't tempted to revert to that "model". But here I think it makes sense I'll update this accordingly.

Reply:
Haha, that's quite the structure.  It feels like there's valid intent behind it (perhaps attempting to _force_ a canonical import path?), but yeah, that's gone pretty overboard.